src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```
@@ -58,7 +58,10 @@ trait Contexts { self: Analyzer =>
   def warnUnusedImports(unit: CompilationUnit) = {
     val imps = allImportInfos(unit).reverse.distinct
 
-    for (imp <- imps) {
+    def languageFeatureImport(imp: ImportInfo): Boolean =
+      imp.qual.sym eq ScalaPackage.info.decl("language": TermName)
```

The problem with the code as written are two-fold: first it will fail silently if `scala.language` is removed (that's what `getRequiredXxx` used in `Definitions` solves); and second you're paying for a name creation and scope lookup for each import.